R/pkg/NAMESPACE

```diff
@@ -363,4 +363,9 @@ S3method(structField, jobj)
 S3method(structType, jobj)
 S3method(structType, structField)
 
+export("newJObject")
+export("callJMethod")
+export("callJStatic")
+export("cleanup.jobj")
```

Comment:
Hi @sun-rui 

Thanks for pointing this out. I look at the code and you are right in mentioning that the jobj has the cleanup.jobj registered. 

Initially, we were under the assumption that we have to register to the the object created otherwise the jvm (running scala) won't cleanup our obj, as there is the ref in the SparkR.

In summary,
 1) For the systemML, with the information you had mentioned, we would be fine with if it is made private also.
 2) Also as you had mentioned, if package writer (on the top of sparkR) feels that they have absolutely run the cleanup, they can call the R's explicit "gc". 

thanks,
Alok